arangod/IResearch/IResearchKludge.cpp

```diff
@@ -112,12 +112,35 @@ void mangleNested(std::string& name) {
 void mangleField(std::string& name, bool isSearchFilter,
                  iresearch::FieldMeta::Analyzer const& analyzer) {
   normalizeExpansion(name);
-  if (isSearchFilter || analyzer._pool->requireMangled()) {
+  if (isSearchFilter) {
     name += kAnalyzerDelimiter;
+
     name += analyzer._shortName;
   } else {
     mangleString(name);
+    if (analyzer->requireMangled()) {
+      name += kAnalyzerDelimiter;
+      name += analyzer._shortName;
+    }
```

Comment:
We discussed and decide didn't do it?
I also think if needed we can achieve it another way (We should have fieldAnalyzerProvider in 'exists', etc)